File: Lab1/SimpleGraphDotNET/DirectedGraph/DirectedVertex.cs
namespace SimpleGraphDotNET.DirectedGraph;

public class DirectedVertex<T>
{
    private readonly List<DirectedVertex<T>> _children;

    public DirectedVertex(T vertexValue)
    {
        VertexValue = vertexValue; 
        _children = new List<DirectedVertex<T>>();
    }

    public T VertexValue { get; set; }
    public IReadOnlyCollection<DirectedVertex<T>> Children => _children;

    public DirectedVertex<T> AddChild(DirectedVertex<T> vertex)
    {
        if (vertex == null) throw new ArgumentNullException(nameof(vertex));
        if (_children.Contains(vertex)) return this;
        _children.Add(vertex);
        return this;
    }
}

File: Lab1/JavaSimpleGraph/src/main/java/DirectedVertex.java
import java.util.ArrayList;
import java.util.List;

public class DirectedVertex<T>
{
    private final List<DirectedVertex<T>> _children;
    public T value;

    public DirectedVertex(T vertexValue)
    {
        value = vertexValue;
        _children = new ArrayList();
    }

    public List<DirectedVertex<T>> getChildren()
    {
        return new ArrayList(_children);
    }

    public DirectedVertex<T> addChild(DirectedVertex<T> vertex)
    {
        if (vertex == null) throw new IllegalArgumentException("Vertex can't be null!");
        if (_children.contains(vertex)) return this;
        _children.add(vertex);
        return this;
    }
}